plugins/statemachineviewer/CMakeLists.txt

```diff
@@ -46,22 +46,8 @@ if(NOT GAMMARAY_CLIENT_ONLY_BUILD)
     endif()
 endif()
 
-# UI part - dependencies
-set(KDSME_PACKAGE_NAME KDSME)
-if(QT_VERSION_MAJOR GREATER_EQUAL 6)
-    set(KDSME_PACKAGE_NAME KDSME-qt6)
-endif()
-set_package_properties(
-    ${KDSME_PACKAGE_NAME} PROPERTIES
-    URL "https://github.com/KDAB/KDStateMachineEditor"
-    DESCRIPTION "KDAB State Machine Editor framework"
-    TYPE RECOMMENDED
-    PURPOSE "Graphical state machine debugging."
-)
-find_package(${KDSME_PACKAGE_NAME} 1.2 CONFIG QUIET)
-
 # UI part
-if(GAMMARAY_BUILD_UI AND ${KDSME_PACKAGE_NAME}_FOUND)
+if(GAMMARAY_BUILD_UI AND GAMMARAY_WITH_KDSME)
```

Comment:
This breaks the ability to use a KDSME package installed separately on the system, doesn't it? Since `find_package()` is no longer even attempted, even if KDSME is installed, it won't be used — the submodule is the only option. And if the `.git` directory isn't present (e.g. package releases), there's no way to make it use a KDSME at all.

Distro packagers won't like that, for one. ...Unless I'm completely missing something?

Reply:
(FTR, my suggestion would be to keep the `find_package()`, but insert a condition to trigger the submodule stuff:

```cmake
if(NOT ${KDSME_PACKAGE_NAME}_FOUND AND GAMMARAY_WITH_KDSME)
```

Maybe even change the option name to `GAMMARAY_USE_INTERNAL_KDSME` or something.

Reply:
Yes, it breaks, I did that because of the includes.
- We could include it on releases, remove/disable the git submodule check, and would also make more sense to static link to it.
- Another option would be to #ifdef the includes
- Fix KDSME includes, which isn't trivial
@winterz @Waqar144 thoughts?

Reply:
A fourth option:

* Keep the submodule, but instead of using it as a build `subdirectory()`, have the CI build-and-install KDSME from there before it configures GammaRay. Then pick it up with `find_package()` as usual.

The built KDSME could even be added to the GitHub cache, to speed things up.

Reply:
Or, in fact, you could use [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html) to incorporate the submodule'd KDSME — with CMake 3.24+, if you give `FetchContent_Declare()` a `FIND_PACKAGE_ARGS` argument, it'll automatically automatically try a `find_package()` first, and use the installed package if it's found.

(`FetchContent_Declare()` doesn't _have_ to download it from a remote repo. Like `ExternalProject_Add()`, you can give it a `SOURCE_DIR` argument with the path to existing sources.)

Reply:
FetchContent doesn't solve the include issue, it's just a little better ExternalProject_Add() but we are avoiding it in favor of submodules.